src/quicklist.c

```diff
@@ -536,6 +599,21 @@ void quicklistAppendZiplist(quicklist *quicklist, unsigned char *zl) {
     quicklist->count += node->count;
 }
 
+/* Create new node consisting of a pre-formed ziplist.
+ * Used for loading RDBs where entire ziplists have been stored
+ * to be retrieved later. */
+void quicklistAppendEntry(quicklist *quicklist, unsigned char *zl, size_t sz) {
```

Comment:
"it would also be nice to document both here, and in the ziplist one that's used by rdb.c, that it takes ownership of the buffer passed to it." not sure I fully understand this part...

Reply:
the caller passes a pointer and the function takes ownership of that memory (i.e. the caller doesn't need to release it, and should actually never access it again).
something like: `zl - the data to add (pointer becomes the responsibility of quicklist)`